src/TBuffer.cpp

```diff
@@ -4413,11 +4422,131 @@ bool TBuffer::processGBSequence(const std::string& bufferData, const bool isFrom
+        // there is no need to tweak pos in THIS case
+
+        return true;
+    } else if (static_cast<quint8>(bufferData.at(pos)) == 0x80 || static_cast<quint8>(bufferData.at(pos)) > 0xFE) {
+        // Invalid as first byte
+        isValid = false;
+        isToUseReplacementMark = true;
+#if defined(DEBUG_BIG5_PROCESSING)
+        qDebug().nospace() << "TBuffer::processBig5Sequence(...) 1-byte sequence as Big5 rejected!";
+#endif
+    } else {
+        // We have two bytes
+        big5SequenceLength = 2;
+        if ((pos + big5SequenceLength - 1) >= len) {
+            // Not enough bytes to process yet - so store what we have and return
+            if (isFromServer) {
+#if defined(DEBUG_BIG5_PROCESSING)
+                qDebug().nospace() << "TBuffer::processBig5Sequence(...) Insufficent bytes in buffer to "
+                                      "complete Big5 sequence, need at least: "
+                                   << big5SequenceLength << " but we currently only have: " << bufferData.substr(pos).length() << " bytes (which we will store for next call to this method)...";
+#endif
+                mIncompleteSequenceBytes = bufferData.substr(pos);
+            }
+            return false; // Bail out
+        } else {
+            // check if second byte range is valid
+            auto val2 = static_cast<quint8>(bufferData.at(pos + 1));
+            if (val2 < 0x40 || (val2 > 0x7E && val2 < 0xA1) || val2 > 0xFE) {
+                // second byte range is invalid
+                isValid = false;
+                isToUseReplacementMark = true;
+            }
+        }
+
+    }
+
+    // At this point we know how many bytes to consume, and whether they are in
+    // the right ranges of individual values to be valid
+
+    if (isValid) {
+        // Try and convert two byte sequence to Unicode using Qts own
+        // decoder - and check number of codepoints returned
+
+        QString codePoint;
+        if (mMainIncomingCodec) {
+            // Third argument is 0 to indicate we do NOT wish to store the state:
+            codePoint = mMainIncomingCodec->toUnicode(bufferData.substr(pos, big5SequenceLength).c_str(), static_cast<int>(big5SequenceLength),
+                                                      nullptr);
+            switch (codePoint.size()) {
+                default:
+                    Q_UNREACHABLE(); // This can't happen, unless we got start or length wrong in std::string::substr()
+                    qWarning().nospace() << "TBuffer::processBig5Sequence(...) " << big5SequenceLength << "-byte Big5 sequence accepted, and it encoded to "
+                                         << codePoint.size() << " QChars which does not make sense!!!";
+                    isValid = false;
+                    isToUseReplacementMark = true;
+                    break;
+                case 2:
+                    // Fall-through
+                    [[clang::fallthrough]];
+                case 1:
```

Comment:
Can you insert after this line:
```
if (codePoint.contains(QChar::ReplacementCharacter)) {
    isValid = false;
    isToUseReplacementMark = true;
    break;
}
```
Having taken a look at the Qt decoder I see that it does put in the replacement character `�` {U+FFFD REPLACEMENT CHARACTER} if it finds a character that it doesn't recognise so that does part of the validation work for us - setting those two flags to those values should activate debug code later on - which, as it happens would also put in that same character but makes this method work the same as it's siblings...